python/housinginsights/ingestion/Manifest.py

```diff
@@ -0,0 +1,135 @@
+        """
+        self.unique_ids = {}
+
+        for row in self:
+            if row['unique_data_id'] in self.unique_ids:
+                return False
+            else:
+                #don't add flags that won't make it into the SQL database
+                if row['include_flag'] in Manifest._include_flags_positive:
+                    self.unique_ids[row['unique_data_id']] = 'found'
+        return True
+
+    def get_manifest_row(self, unique_data_id):
+        """
+        Returns the row for the given unique_data_id else 'None' if not in
+        manifest.
+        """
+        for row in self:
+            use = row['include_flag'] == 'use'
+            if row['unique_data_id'] == unique_data_id and use:
+                return row
+        return None
+
+    def create_list(self, folder_path):
+        """
+        Returns a list of potential unique_data_ids based on csv files in the
+        given folder path. The assumption is that all raw data a saved as csv
+        with filename that matches unique_data_id used in manifest.
+        """
+        unique_data_ids = list()
+        files_in_folder = os.listdir(folder_path)
+        for file in files_in_folder:
+            file_path = os.path.join(folder_path, file)
+            uid, file_ext = file.split(sep='.')
+            if os.path.isfile(file_path) and file_ext == 'csv':
+                unique_data_ids.append(uid)
+
+        return unique_data_ids
+
+    def update_manifest(self, date_stamped_folder):
+        timestamp = os.path.basename(date_stamped_folder)
+        data_date = datetime.strptime(timestamp, '%Y%m%d').strftime('%m/%d/%Y')
+        file_path_base = date_stamped_folder[date_stamped_folder.find('raw'):]
+
+        field_names = self.keys
+
+        # get unique_data_ids for data files in recent subfolder
+        uid_list = self.create_list(date_stamped_folder)
+        data = list()
+
+        # update the manifest.csv in place
+        for row in self:
+            row_uid = row['unique_data_id']
+            if row_uid in uid_list:
+                row['data_date'] = data_date
+                filepath = os.path.join(file_path_base,
+                                        "{}.csv".format(row_uid))
+                row['filepath'] = filepath
+            else:
+                # parse the date back into format used in manifest
+                # TODO - discuss with Neal to standardize timestamp in
+                # TODO - manifest so we don't have to clean between sql and
+                # TODO - manifest
+                try:
+                    _date = dateparser.parse(row['data_date'], dayfirst=False,
+                                             yearfirst=False)
+                    row['data_date'] = datetime.strftime(_date, '%m/%d/%Y')
+                except ValueError:
+                    row['data_date'] = 'Null'
+            data.append(row)
+
+        with open(self.path, mode='w', encoding='utf-8') as f:
```

Comment:
Note, when I ran this it produced extra lines. We've encountered this in other places in the code too. The solution is to open with `newline=''` i.e. 
`with open(self.path, mode='w', encoding='utf-8', newline='') as f:`

I've made this change and will push it to dev when finished testing locally.